smart_open/gcs.py

```diff
@@ -399,6 +400,8 @@ def __init__(
 
         self._total_size = 0
         self._total_parts = 0
+        self._bytes_uploaded = 0
+        self._closed = False
```

Comment:
Do we really need this as a separate variable? Can we not use `self._client is None` to mean we're closed, and assume open otherwise?

Reply:
Once we have uploaded the final part (signified by a content range with an end), any additional put requests lead to a hanging response that takes a while to timeout. Saving state of the completeness of the upload was done to avoid this hanging response.

Reply:
OK, then please set `self._client = None` instead of using `self._closed = True`.